src/test/java/org/apache/sling/resourceresolver/impl/mapping/MapEntriesTest.java

```diff
@@ -168,6 +172,8 @@ public void test_simple_alias_support() {
         });
 
         mapEntries.doInit();
+        
+        while(!mapEntries.isAliasMapInitialized()){}
```

Comment:
An empty loop looks wrong. I know what it does :-) but it looks wrong. I would suggest either adding a method to mapEntries for waiting for it to be available or using a test helper method.

Also please add timeouts to all methods having such async logic to prevent them from hanging indefinitely under special circumstances.

Reply:
@rombert  - I'll be adding timeout to the Test annotation - this is really kind of a test hack  to wait for this to complete. I can't see any use case for waiting for the alias map in reality so I'm a bit hesitant to add logic to the MapEntries class just to trigger some waiting test. I'll be checking if there is some mockito helper to watch out for a certain condition to change.